### PR TITLE
chore: fix security alerts

### PR DESCRIPTION
**Pull request by rchincha**
https://github.com/project-zot/zot/security/code-scanning/4293

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -461,3 +461,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.44.271
```
Comment (andaaron) [inline go.mod:465]: Why is this needed?

Comment (rchincha) [reply go.mod:465]: https://github.com/project-zot/zot/security/code-scanning/4293

That is not enough because go.sum still has an older version via an indirect dependency. This forces to move everywhere.

Comment (rchincha) [reply go.mod:465]: IINM, `require` specifies a direct dependency but does not guarantee that all indirect ones respect that also. You can try it yourself. Vuln scanners look at go.sum also btw.

Comment (andaaron) [reply go.mod:465]: Actually this go.mod is already up to date on main. Why would we be vulnerable?

Comment (rchincha) [reply go.mod:465]: https://github.com/project-zot/zot/blob/main/go.sum#L400

Comment (andaaron) [reply go.mod:465]: That is a list of possible versions to use, not the version to be included in the binary at build time. v1.44.271 is there, since we have another higher version - that one should be used https://github.com/project-zot/zot/blob/main/go.sum#L403.

Comment (rchincha) [reply go.mod:465]: $ go mod graph  | grep "github.com/aws/aws-sdk-go@v1.20.6"
github.com/apex/log@v1.9.0 github.com/aws/aws-sdk-go@v1.20.6

$ go mod graph | grep "github.com/apex/log@v1.9.0"
github.com/opencontainers/umoci@v0.4.8-0.20210922062158-e60a0cc726e6 github.com/apex/log@v1.9.0

^ here is the actual dep graph

This is the whole point of go.sum

Comment (andaaron) [reply go.mod:465]: Yes. but only a single version of the library is used at compile time, the maximum of all the minim versions specified as dependencies.

Comment (andaaron) [reply go.mod:465]: From the link above
```
MVS starts at the main modules (special vertices in the graph that have no version) and traverses the graph, tracking the highest required version of each module. At the end of the traversal, the highest required versions comprise the build list: they are the minimum versions that satisfy all requirements.
```

Comment (rchincha) [reply go.mod:465]: https://go.dev/ref/mod#minimal-version-selection

yes, but this or we find a way around the vuln. scanner

**Review by andaaron — APPROVED** [review]
